lib/ds/RequestExecutor.js

```diff
@@ -84,7 +84,7 @@ RequestExecutor.prototype.execute = function executeRequest(req, callback) {
     }
 
     if (response.statusCode > 399) {
-      return callback(new ResourceError(body || {status:response.statusCode}), null);
+      return callback(new ResourceError(body || {status:response.statusCode}, {url: req.uri, method: req.method}), null);
```

Comment:
I do want to have the full URL logged, can you show me how you produced the `/tenants/current` value? When I tested this, I simply made a request for an application that doesn't exist:

```
var stormpath = require('stormpath');

var client = new stormpath.Client();

var applicationHref = 'https://api.stormpath.com/v1/applications/foo';

client.getResource(applicationHref,function (err) {
  console.log(err);
});
```

And I get this error, which gives me the full URL:

```
{ [ResourceError: HTTP 404, Stormpath 404 (mailto:support@stormpath.com): The requested resource does not exist.]
  name: 'ResourceError',
  status: 404,
  code: 404,
  userMessage: 'The requested resource does not exist.',
  developerMessage: 'The requested resource does not exist.',
  moreInfo: 'mailto:support@stormpath.com',
  requestId: '8d80cd20-534b-11e6-a56f-22000bc0820c',
  url: 'https://api.stormpath.com/v1/applications/foo',
  method: 'GET',
  message: 'HTTP 404, Stormpath 404 (mailto:support@stormpath.com): The requested resource does not exist.' }
```